coreneuron/io/reports/report_event.hpp

```diff
@@ -20,15 +20,15 @@ namespace coreneuron {
 
 #if defined(ENABLE_BIN_REPORTS) || defined(ENABLE_SONATA_REPORTS)
 struct VarWithMapping {
-    int id;
+    uint32_t id;
     double* var_value;
     VarWithMapping(int id_, double* v_)
```

Comment:
Also I see the constructor being call with `int`s, so is this something that needs to be changed in all the calls to `VarWithMapping`?
I see the `section_id`s being `int`s around the code also. Is this because we pass these as `int`s from neurodamus? Should this be changed as well?

Reply:
We would have to change everywhere (also in neurodamus for the gids)
For now i just changed the obvious places before we call the libsonata-report functions

Do you think we should change it everywhere? @pramodk  do you have an opinion?

Reply:
i think this is neuron legacy coming at us, but really for all these ids it would make much more sense to work with unsigned.